Classes/Configuration/BackendConfigurationManager.php

```diff
@@ -110,7 +110,7 @@ protected function getPageIdFromRecordIdentifiedInEditUrlArgument() {
 		if ('pages' === $table && 'new' === $command) {
 			// if TYPO3 wants to insert a new page, URL argument is already the PID value.
 			return $id;
-		} elseif (0 <> $id) {
+		} elseif (0 <> $id && 'pages' !== $table) {
```

Comment:
What if line 110 instead removes the `&& 'new' === $command` check? That'd cover this case, too. There's also the consideration: what happens when `$id` is a negative value (e.g. insert after)?